pkg/producer/limit.go

```diff
@@ -0,0 +1,28 @@
+package producer
+
+import (
+	"context"
+	"encoding/binary"
+
+	"github.com/asecurityteam/nexpose-vuln-hydrator/pkg/domain"
+)
+
+// SizeLimitProducer limits the size of events that can be passed
+// to an underlying producer.
+type SizeLimitProducer struct {
+	SizeLimit int
+	Wrapped   domain.Producer
+}
+
+// Produce checks the size of an event, and returns an error if it exceeds the configured limit.
+// Otherwise, it passes the event through to the wrapped producer.
+func (p *SizeLimitProducer) Produce(ctx context.Context, event interface{}) (interface{}, error) {
+	size := binary.Size(event)
```

Comment:
Question: should the struct be marshaled to JSON first to get a more accurate picture of whether the SQS payload will fail?

Reply:
Will add that

Reply:
Fixed.